lib/rules/array-bracket-newline.js

```diff
@@ -15,8 +15,26 @@ const astUtils = require("./utils/ast-utils");
 /** @type {import('../shared/types').Rule} */
 module.exports = {
     meta: {
-        deprecated: true,
-        replacedBy: [],
+        deprecated: {
+            message: "Stylistic rules are being moved out of ESLint core.",
+            url: "https://eslint.org/blog/2023/10/deprecating-formatting-rules/",
+            deprecatedSince: "8.53.0",
+            availableUntil: "10.0.0",
+            replacedBy: [
+                {
+                    message: "Maintains now deprecated stylistic core rules",
```

Comment:
```suggestion
                    message: "ESLint Stylistic maintains now deprecated stylistic core rules",
```

Since there's no other text (just URLs) for this replacement entry, I think we should mention the project's name in this message.